### PR TITLE
Build cargo-fuzz (as E2E test)

### PR DESCRIPTION
**Pull request by maxammann**
Adds a cargo-fuzz example fuzzer to demonstrate how to use the libfuzzer compatibility layer. It also acts as a E2E test.

Tries to reproduce https://github.com/AFLplusplus/LibAFL/issues/2039

### CODE REVIEW

**Comment by tokatoka** [comment]
we changed a lot of stuff in CI lately, and the failure in 1 fuzzer example will cancel all the runs for other fuzzers

**Comment by tokatoka** [comment]
so... it is working in the end right?

**Comment by tokatoka** [comment]
i cancelled it because it's running forever

File: fuzzers/cargo_fuzz/fuzz/fuzz_targets/fuzz_target_1.rs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#![no_main]
+
+use libfuzzer_sys::fuzz_target;
+
+fuzz_target!(|data: &[u8]| {
+    // fuzzed code goes here
```
Comment (tokatoka) [inline fuzzers/cargo_fuzz/fuzz/fuzz_targets/fuzz_target_1.rs:6]: maybe add something here?

File: fuzzers/cargo_fuzz/Makefile.toml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+dependencies = [ "install_cargo_fuzz", "install_llvm_tools"]
+
+[tasks.test]
+linux_alias = "test_unix"
+mac_alias = "unsupported"
+windows_alias = "unsupported"
+
+[tasks.test_unix]
+script='''
+'''
```
Comment (tokatoka) [inline fuzzers/cargo_fuzz/Makefile.toml:31]: what we usually do is. 
run the fuzzer for 30 secs, grep the stdout, find the pattern "corpus: XXX".
if found then success, else no.

**Comment by addisoncrump** [comment]
Completed here: #2081

**Pull request closed without merging**